integration-tests/cypress/e2e/project/react-flow/diagram-contextual-palette.cy.js

```diff
@@ -164,4 +164,42 @@ describe('/projects/:projectId/edit - Robot Diagram', () => {
         cy.getByTestId('tool2_section2 - Tool').should('not.exist');
       });
   });
+
+  it('remembers the last tool invoked from a tool section', () => {
+    // Create a studio project with a Domain diagram
+    cy.createProjectFromTemplate('studio-template').then((res) => {
+      const projectId = res.body.data.createProjectFromTemplate.project.id;
+      cy.visit(`/projects/${projectId}/edit`);
+    });
+
+    // Rename the diagram switch it to React Flow
+    cy.getByTestId('onboard-open-Domain').click();
+    cy.getByTestId('Domain').click();
+    cy.getByTestId('Domain-more').click();
+    cy.getByTestId('rename-tree-item').click();
+    cy.getByTestId('name-edit').type('Domain__REACT_FLOW{enter}');
+    // Close and re-open to be sure
+    cy.getByTestId('close-representation-tab-Domain__REACT_FLOW').click();
+    cy.getByTestId('Domain__REACT_FLOW').click();
```

Comment:
I'm not sure it's necessary, I've never had this happen with a cypress test.

Reply:
Removed.